src/components/Roles/AddRoleDrawer.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  Button,
+  Drawer,
+  DrawerBody,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerHeaderTitle,
+  Label,
+  MessageBar,
+  MessageBarBody,
+  MessageBarTitle,
+  Option,
+  Spinner,
+  Text,
+} from "@fluentui/react-components";
+import { DismissRegular } from "@fluentui/react-icons";
+import {
+  Controller,
+  FormProvider,
+  SubmitHandler,
+  useForm,
+} from "react-hook-form";
+import { useEffect } from "react";
+import { ContactIcon, SaveIcon } from "@fluentui/react-icons-mdl2";
+import { NewRole, useAddRole } from "api/rolesApi";
+import { ROLES } from "consts/Roles";
+import BACCombobox from "components/BaseFormFields/BACCombobox";
+import { PeoplePicker } from "components/PeoplePicker/PeoplePicker";
+
+const AddRoleDrawer = ({
+  isOpen,
+  setIsOpen,
+}: {
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+}) => {
+  const addRole = useAddRole();
+
+  const myForm = useForm<NewRole>({
+    criteriaMode:
+      "all" /* Pass back multiple errors, so we can prioritize which one(s) to show */,
+    mode: "onChange" /* Provide input directly as they input, so if entering bad data (eg letter in MPCN) it will let them know */,
+  });
+
+  // Reset the form when form is opened/closed
+  useEffect(() => {
+    myForm.reset();
+  }, [isOpen, myForm]);
```

Comment:
Just pass the reset function into the dependency array instead of entire object?
```suggestion
  }, [isOpen, myForm.reset]);
```